### PR TITLE
Feature / Artemis ID

### PR DESCRIPTION
**Pull request by jfk056**

### CODE REVIEW

File: Sources/RingPublishingTracking/Private/Networking/Endpoints/Responses/ArthemisResponse.swift [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+        let id: ID
+    }
+
+    struct ID: Decodable {
+        let real: String
+
+        let model: String
+
+        let models: Models
```
Comment (adamszeremeta) [inline Sources/RingPublishingTracking/Private/Networking/Endpoints/Responses/ArthemisResponse.swift]: I just wonder if this should be handled as blackbox or not - I will aks Michał for confirmation

File: Sources/RingPublishingTracking/Private/Networking/Endpoints/Responses/ArthemisResponse.swift [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            case atsRi = "ats_ri"
+        }
+    }
+
+    let cfg: ArtemisResponse.CFG
+
+    let user: ArtemisResponse.User
+
+    func transform() -> ArtemisObject {
+        let external: ArtemisObject.ID.External = .init(model: user.id.model, models: user.id.models.atsRi)
```
Comment (adamszeremeta) [inline Sources/RingPublishingTracking/Private/Networking/Endpoints/Responses/ArthemisResponse.swift]: I would always write it as ArtemisObject.ID.External(model: user.id.model, models: user.id.models.atsRi)

File: Sources/RingPublishingTracking/Private/Tracking/Events/EventsService.swift [hunk 3]
```diff
@@ -84,11 +100,9 @@ final class EventsService {
 
     /// Setups API Service
     /// - Parameters:
-    ///   - apiUrl: API url. If not provided the default URL will be used
-    ///   - apiKey: API key
     ///   - session: Session object used to create requests. Defaults to `URLSession.shared`
-    func setup(apiUrl: URL?, apiKey: String, delegate: EventsServiceDelegate?, session: NetworkSession = URLSession.shared) {
-        apiService = APIService(apiUrl: apiUrl ?? Constants.apiUrl, apiKey: apiKey, session: session)
+    func setup(delegate: EventsServiceDelegate?, session: NetworkSession = URLSession.shared) {
```
Comment (adamszeremeta) [inline Sources/RingPublishingTracking/Private/Tracking/Events/EventsService.swift:105]: missing delegate param in method comment

File: Sources/RingPublishingTracking/Private/Tracking/Events/EventsService.swift [hunk 4]
```diff
@@ -110,9 +124,33 @@ final class EventsService {
...
+                    self.publishTrackingIdentifier(eaUUID: identifiers.0, artemis: identifiers.1)
+                    self.isIdentifyMeRequestInProgress = false
+                case .failure(let error):
+                    switch self.shouldRetryIdentifyRequest {
+                    case true:
+                        self.handleIdentifyMeRequestFailure(error: error)
+                        self.isIdentifyMeRequestInProgress = false
+                    case false:
+                        guard let eaUUID = storage.eaUUID else { return }
+                        self.retryArtemisRequest(eaUUID: eaUUID) { artemisResult in
```
Comment (adamszeremeta) [inline Sources/RingPublishingTracking/Private/Tracking/Events/EventsService.swift]: maybe we could flatten it a little by extracting to other helper methods?

File: Sources/RingPublishingTracking/Public/Model/TrackingIdentifier.swift [hunk 5]
```diff
@@ -11,9 +11,25 @@ import Foundation
 /// Tracking identifier and its expiration date
 public struct TrackingIdentifier {
 
-    /// Identifier
-    public let identifier: String
+    public struct EaUUID {
```
Comment (adamszeremeta) [inline Sources/RingPublishingTracking/Public/Model/TrackingIdentifier.swift]: could we have those models in separate files? just public stuff, codable entities can be like you did them

File: Sources/RingPublishingTracking/Private/Networking/Models/Artemis.swift [hunk 6]
```diff
@@ -0,0 +1,39 @@
+//
+//  ArtemisID.swift
+//  RingPublishingTracking
+//
+//  Created by Adam Mordavsky on 09.11.23.
+//
+
+import Foundation
+
+public struct ArtemisObject: Codable {
```
Comment (adamszeremeta) [inline Sources/RingPublishingTracking/Private/Networking/Models/Artemis.swift]: should not be public (as it is in private directory), better to have separate public model

Comment (adamszeremeta) [reply Sources/RingPublishingTracking/Private/Networking/Models/Artemis.swift]: and for public stuff, each property and object should have a comment saying that it is

**Review by adamszeremeta — CHANGES_REQUESTED** [review]
We would need also:
- swiftlint fix (https://bamboo.grupa.onet/browse/MIOS-APTRACK48-2)
- changelog added (as 1.4.0)
- bumper project version to 1.4.0 (demo project)
- and this RDLU param extended with artemis (as I don't see it yet here?)

File: Sources/RingPublishingTracking/Private/Decorator/Generic/UserDataDecorator.swift [hunk 7]
```diff
@@ -34,7 +41,20 @@ final class UserDataDecorator: Decorator {
...
+
+    func updateArtemisData(artemis: ArtemisID?) {
+        data?.id = artemis
+    }
+}
+
+private extension UserDataDecorator {
+
+    func prepareRDLU(data: UserData?) -> Data? {
+        guard let data = data, data.sso != nil || data.id != nil else { return nil }
```
Comment (danielcalka) [inline Sources/RingPublishingTracking/Private/Decorator/Generic/UserDataDecorator.swift:56]: Tutaj mamy chyba rozbieżność z androidem - u nas dodajemy RDLU tylko jeśli artemis != null. Tutaj możemy dodać dla samego sso. Która wersja jest ok?

**Review by danielcalka — APPROVED** [review]

**Review by adamszeremeta — APPROVED** [review]

**Pull request merged**